### PR TITLE
More failure codepath tracing wiring

### PR DESCRIPTION
**Pull request by ryoqun**
related to this #15237 , another some very low-risk niceties, this is needed when I was doing last periodic inflation verification.

There should be no functional change.

### CODE REVIEW

**Comment by ryoqun** [comment]
@CriesofCarrots could you sanity-check this at your free time? :)

File: programs/stake/src/stake_state.rs [hunk 1]
```diff
@@ -592,8 +614,14 @@ impl Stake {
         // if there is no newer credits since observed, return no point
         if new_vote_state.credits() <= self.credits_observed {
             if fix_stake_deactivate {
+                if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
+                    inflation_point_calc_tracer(&SkippedReason::ZeroCreditsAndReturnCurrent.into());
+                }
                 return (0, self.credits_observed);
             } else {
+                if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
```
Comment (CriesofCarrots) [inline programs/stake/src/stake_state.rs:622]: We shouldn't ever be hitting this case, right?

**Review by CriesofCarrots — APPROVED** [review]
lgtm!

File: programs/stake/src/stake_state.rs (hunk 1 above)
Comment (ryoqun) [reply programs/stake/src/stake_state.rs:622]: Right, we shouldn't ever be hitting. :) That's one of the things I actually want to reaffirm, if there should be any oddities in the distant future. :)

Comment (ryoqun) [reply programs/stake/src/stake_state.rs:622]: (Also, I think we're nearing to the phase where we can remove this dead altogether safely)

**Comment by ryoqun** [comment]
now that another v1.5 is released; time to trickle down this.

**Pull request merged**